src/main/java/org/jboss/msc/service/ObsoleteServiceBuilderImpl.java

```diff
@@ -0,0 +1,357 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2010, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.msc.service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.msc.inject.Injector;
+import org.jboss.msc.inject.Injectors;
+import org.jboss.msc.value.ImmediateValue;
+import org.jboss.msc.value.Value;
+
+/**
+ * Obsolete {@link ServiceBuilder} implementation.
+ *
+ * @param <T> the type of service being built
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ * @author <a href="mailto:ropalka@redhat.com">Richard Opalka</a>
+ */
+class ObsoleteServiceBuilderImpl<T> extends AbstractServiceBuilder<T> {
```

Comment:
What is the purpose of having a separate service builder here?

Reply:
Two main reasons:
 * every code change has a potential to introduce new bugs - so in order to minimize this
   I decided to maintain old service builder implementation and throw 
   UnsupportedOperationException in requires(), provides() and setInstance() methods
 * With introduction of MSC-203 there are two MSC APIs now merged into one:
   [1] obsolete MSC API - the one we deprecated (includes "old" injectors / values framework)
   [2] new MSC API - based just on requires(), provides() and setInstance() methods

The implementation detail is I decided to don't allow MSC API consumers to mix these two APIs
[1] & [2] in one service builder. There is clear separation of them. See detailed commit message of MSC-203 implementation.

Reply:
OK this makes sense, thanks.